include/nbl/asset/filters/CNormalMapToDerivativeFilter.h

```diff
@@ -0,0 +1,275 @@
+				size_t	scratchMemoryByteSize = {};											//!< required byte size for entire scratch memory
+
+				static inline size_t getRequiredScratchByteSize(asset::VkExtent3D extent)
+				{
+					size_t retval = extent.width * extent.height * extent.depth * decodeTypeByteSize * forcedScratchChannelAmount;
+					
+					return retval;
+				}
+
+				/*
+					Layer ID is relative to outBaseLayer in state
+				*/
+
+				const std::array<double, forcedScratchChannelAmount>& getAbsoluteLayerScaleValue(size_t layer)
+				{
+					if (!maxAbsLayerScaleValues.empty())
+						return maxAbsLayerScaleValues[layer];
+					else
+						return {};
+				}
+
+			protected:
+				std::vector<std::array<double, forcedScratchChannelAmount>> maxAbsLayerScaleValues; 								//!< scales gained by the filter (each layer handled) for derivative map shader usage
+		};
+
+	protected:
+		CNormalMapToDerivativeFilterBase() {}
+		virtual ~CNormalMapToDerivativeFilterBase() {}
+
+		static inline bool validate(CNormalMapToDerivativeStateBase* state)
+		{
+			if (!state)
+				return false;
+
+			return true;
+		}
+};
+
+//! Convert Normal Map to Derivative Normal Map
+/*
+	
+*/
+
+class CNormalMapToDerivativeFilter : public CMatchedSizeInOutImageFilterCommon, public CNormalMapToDerivativeFilterBase
+{
+	public:
+		virtual ~CNormalMapToDerivativeFilter() {}
+
+		class CStateBase : public CMatchedSizeInOutImageFilterCommon::state_type, public CNormalMapToDerivativeFilterBase::CNormalMapToDerivativeStateBase
+		{ 
+			public:
+				CStateBase() = default;
+				virtual ~CStateBase() = default;
+
+			private:
+
+				void resetLayerScaleValues()
+				{
+					maxAbsLayerScaleValues.clear();
+				}
+
+				friend class CNormalMapToDerivativeFilter;
+		};
+		using state_type = CStateBase; //!< full combined state
+
+		static inline bool validate(state_type* state)
+		{
+			if (!CMatchedSizeInOutImageFilterCommon::validate(state))
+				return false;
+
+			if (!CNormalMapToDerivativeFilterBase::validate(state))
+				return false;
+			
+			const ICPUImage::SCreationParams& inParams = state->inImage->getCreationParameters();
+			const ICPUImage::SCreationParams& outParams = state->outImage->getCreationParameters();
+			const auto inFormat = inParams.format;
+			const auto outFormat = outParams.format;
+
+			if (outFormat != asset::EF_R8G8_SNORM)
+				return false;
+
+			if (state->scratchMemoryByteSize < state_type::getRequiredScratchByteSize(state->extent))
+				return false;
+
+			if (asset::getFormatChannelCount(inFormat) < 3 )
+				return false;
+
+			return true;
+		}
+
+		static inline bool execute(state_type* state)
+		{
+			if (!validate(state))
+				return false;
+
+			state->resetLayerScaleValues();
+
+			auto checkFormat = state->inImage->getCreationParameters().format;
+			if (isIntegerFormat(checkFormat))
+				return executeInterprated(state, reinterpret_cast<uint64_t*>(state->scratchMemory));
```

Comment:
no point supporting integer formats, only normalized or full floats

also fix the spelling on `executeInterprated`

Reply:
corrected in following commit